docs/upgrade-to-v5.md

```diff
@@ -130,6 +130,36 @@ var groups = await graphServiceClient
     });
 ```
 
+### Per-Request Options
+To pass per-request options to the default http middleware to configure actions like redirects and retries, this can be done using the `requestConfiguration` by adding an `IRequestOption` instance to the `Options` collection. For example, adding a `RetryHandlerOption` instance to configure the retry handler option as below.
+
+```cs
+
+var retryHandlerOption = new RetryHandlerOption
+{
+    MaxRetry = 7,
+    ShouldRetry = (delay,attempt,message) => true
+};
+var user = await graphClient.Me.GetAsync(requestConfiguration => requestConfiguration.Options.Add(retryHandlerOption));
+```
+
+Other `IRequestOption` instances provided by default include the following and their source can be found [here](https://github.com/microsoft/kiota-http-dotnet/tree/main/src/Middleware/Options)
+
+- RetryHandlerOption - for configuring the retry handler to customise request retries
```

Comment:
All options should be marked as code like `RetryHandlerOption`, `RedirectHandlerOption` and `ChaosHandlerOption`